sway-lsp/src/capabilities/hover/hover_link_contents.rs

```diff
@@ -95,37 +94,31 @@ impl<'a> HoverLinkContents<'a> {
 
     /// Adds all implementations of the given [`TyTraitDecl`] to the list of implementations.
     pub fn add_implementations_for_trait(&mut self, trait_decl: &TyTraitDecl) {
-        if let Some(namespace) = self.session.namespace() {
-            let call_path =
-                CallPath::from(trait_decl.name.clone()).to_fullpath(self.engines, &namespace);
-            let impl_spans =
-                TraitMap::get_impl_spans_for_trait_name(namespace.current_module(), &call_path);
-            self.add_implementations(&trait_decl.span(), impl_spans);
-        }
+        let call_path =
+            CallPath::from(trait_decl.name.clone()).to_fullpath(self.engines, self.namespace);
+        let impl_spans =
+            TraitMap::get_impl_spans_for_trait_name(self.namespace.current_module(), &call_path);
+        self.add_implementations(&trait_decl.span(), impl_spans);
     }
 
     /// Adds implementations of the given type to the list of implementations using the [`TyDecl`].
     pub fn add_implementations_for_decl(&mut self, ty_decl: &TyDecl) {
-        if let Some(namespace) = self.session.namespace() {
-            let impl_spans = TraitMap::get_impl_spans_for_decl(
-                namespace.current_module(),
-                self.engines,
-                ty_decl,
-            );
-            self.add_implementations(&ty_decl.span(self.engines), impl_spans);
-        }
+        let impl_spans = TraitMap::get_impl_spans_for_decl(
+            self.namespace.current_module(),
+            self.engines,
+            ty_decl,
+        );
+        self.add_implementations(&ty_decl.span(self.engines), impl_spans);
```

Comment:
nit: `add_implementations` has `all_spans.dedup();`; this only removes adjacent duplicates in a vector; hence not guaranteed to remove all duplicates if vec is unsorted.
May need to sort vec, or use hashset instead.

Reply:
updated